org-code-javabuilder/theater/src/main/java/org/code/theater/Scene.java

```diff
@@ -0,0 +1,436 @@
+package org.code.theater;
+
+import static org.code.theater.support.DrawImageAction.UNSPECIFIED;
+
+import java.io.FileNotFoundException;
+import java.util.ArrayList;
+import java.util.List;
+import org.code.media.*;
+import org.code.theater.support.*;
+
+public class Scene {
+  private static final int WIDTH = 400;
+  private static final int HEIGHT = 400;
+
+  // Visible for testing
+  static final Font DEFAULT_FONT = Font.MONO;
+  static final FontStyle DEFAULT_FONT_STYLE = FontStyle.NORMAL;
+  static final int DEFAULT_TEXT_HEIGHT = 20;
+  static final Color DEFAULT_COLOR = Color.BLACK;
+  static final double DEFAULT_STROKE_WIDTH = 1.0;
+
+  private final List<SceneAction> actions;
+
+  private Font font;
+  private FontStyle fontStyle;
+  private int textHeight;
+  private Color textColor;
+  private Color strokeColor;
+  private Color fillColor;
+  private double strokeWidth;
+
+  public Scene() {
+    this.actions = new ArrayList<>();
+
+    this.font = DEFAULT_FONT;
+    this.fontStyle = DEFAULT_FONT_STYLE;
+    this.textHeight = DEFAULT_TEXT_HEIGHT;
+    this.textColor = DEFAULT_COLOR;
+    this.strokeColor = DEFAULT_COLOR;
+    this.fillColor = DEFAULT_COLOR;
+    this.strokeWidth = DEFAULT_STROKE_WIDTH;
+  }
+
+  /** Returns the width of the theater canvas. */
+  public final int getWidth() {
+    return Scene.WIDTH;
+  }
+
+  /** Returns the height of the theater canvas. */
+  public final int getHeight() {
+    return Scene.HEIGHT;
+  }
+
+  /**
+   * Clear the canvas and set the background to the given color name.
+   *
+   * @param color new background color name.
+   */
+  public final void clear(String color) {
+    this.clear(new Color(color));
+  }
+
+  /**
+   * Clear the canvas and set the background to the given color.
+   *
+   * @param color new background color
+   */
+  public final void clear(Color color) {
+    this.actions.add(new ClearSceneAction(color));
+  }
+
+  /**
+   * Plays the array of samples provided.
+   *
+   * @param sound an array of samples to play.
+   */
+  public final void playSound(double[] sound) {
+    this.actions.add(new PlaySoundAction(sound));
+  }
+
+  /**
+   * Plays the sound referenced by the file name.
+   *
+   * @param filename the file to play in the asset manager.
+   */
+  public final void playSound(String filename) {
+    try {
+      this.playSound(AudioUtils.readSamplesFromAssetFile(filename));
+    } catch (FileNotFoundException e) {
+      throw new TheaterRuntimeException(ExceptionKeys.FILE_NOT_FOUND, e);
+    }
+  }
+
+  /**
+   * Plays a note with the selected instrument.
+   *
+   * @param instrument the instrument to play.
+   * @param note the note to play. 60 represents middle C on a piano.
+   * @param seconds length of the note.
+   */
+  public final void playNote(Instrument instrument, int note, double seconds) {
```

Comment:
NIT: Should we also do the `playNote` and `playNoteAndPause` without an instrument? Thinking about the discussion here: https://codedotorg.slack.com/archives/C01EF4GJ9GE/p1647005749541949?thread_ts=1646959030.811979&cid=C01EF4GJ9GE

Reply:
Oh good call, I'll add these